// pug-angularjs-templates-brunch/lib/index.js
// Generated by CoffeeScript 1.7.1
var AngularTemplatesCompiler, escape, pug;

escape = require('js-string-escape');

pug = require('pug');

module.exports = AngularTemplatesCompiler = (function() {
  AngularTemplatesCompiler.prototype.brunchPlugin = true;

  AngularTemplatesCompiler.prototype.type = 'template';

  AngularTemplatesCompiler.prototype.extension = 'pug';

  AngularTemplatesCompiler.prototype.pattern = /\.(jade|pug)/;

  AngularTemplatesCompiler.prototype._default_path_transform = function(path) {
    return path;
  };

  function AngularTemplatesCompiler(config) {
    var _ref, _ref1, _ref2, _ref3, _ref4, _ref5, _ref6, _ref7, _ref8, _ref9;
    this.module = ((_ref = config.plugins) != null ? (_ref1 = _ref.pug_angular_templates) != null ? _ref1.module : void 0 : void 0) || 'templates';
    this.path_transform = ((_ref2 = config.plugins) != null ? (_ref3 = _ref2.pug_angular_templates) != null ? _ref3.path_transform : void 0 : void 0) || this._default_path_transform;
    this.locals = ((_ref4 = config.plugins) != null ? (_ref5 = _ref4.pug_angular_templates) != null ? _ref5.locals : void 0 : void 0) || {};
    this.pretty = !!((_ref6 = config.plugins) != null ? (_ref7 = _ref6.pug_angular_templates) != null ? _ref7.pretty : void 0 : void 0);
    this.doctype = ((_ref8 = config.plugins) != null ? (_ref9 = _ref8.pug_angular_templates) != null ? _ref9.doctype : void 0 : void 0) || "5";
  }

  AngularTemplatesCompiler.prototype.compile = function(data, path, callback) {
    var html, pugfunction, url;
    pugfunction = pug.compile(data, {
      debug: false,
      pretty: this.pretty,
      doctype: this.doctype,
      filename: path,
      compileDebug: false
    });
    html = pugfunction(this.locals);
    html = escape(html);
    url = this.path_transform(path.replace(/\\/g, "/"));
    return callback(null, "(function() {\n    var module;\n\n    try {\n        // Get current templates module\n        module = angular.module('" + this.module + "');\n    } catch (error) {\n        // Or create a new one\n        module = angular.module('" + this.module + "', []);\n    }\n\n    module.run([\"$templateCache\", function($templateCache) {\n        $templateCache.put('" + url + "', '" + html + "');\n    }]);\n})();");
  };

  return AngularTemplatesCompiler;

})();
